src/simulators/statevector/chunk/cuStateVec_chunk_container.hpp

```diff
@@ -20,6 +20,8 @@
 
 #include "custatevec.h"
 
+#define SUPPORTS_MEMORY_POOL ( __CUDACC_VER_MAJOR__ > 11 || (__CUDACC_VER_MAJOR__ == 11 && __CUDACC_VER_MINOR__ >= 2))
```

Comment:
Can we always use memory pool if it provides performance in any simulation? The latest cuQuantum changed its API (such as `custatevecSamplerCreate`) and the current `cuStateVec_chunk_container.hpp` does not support previous API (`custatevecSampler_create`). I guess, if the current code is based on the latest API,  memory pool is always available.

Reply:
I removed codes for older version of CUDA and now 11.2 or newer version is required to use cuStateVec support (added in CONTRIBUTING.md)

I think sampling does not support memory pool (https://docs.nvidia.com/cuda/cuquantum/custatevec/api/functions.html#sampling)
so work buffer should be allocated if needed.